lib/profile_helpers.js

```diff
@@ -20,211 +20,211 @@ function merge_modifiers(data, modifiers) {
 };
 exports.merge_modifiers = merge_modifiers;
 
-// used internally by set and set_once
-// identifiers is one of {$distinct_id} or {$group_key, $group_id}
-exports.create_profile_helpers = function({token, config, send_request}){
-    return {
-        _set: function(prop, to, modifiers, callback, {identifiers, set_once = false, endpoint}) {
-            let $set = {};
-
-            if (typeof(prop) === 'object') {
-                if (typeof(to) === 'object') {
-                    callback = modifiers;
-                    modifiers = to;
-                } else {
-                    callback = to;
-                }
-                $set = prop;
+exports.ProfileHelpers = (Base = Object) => class extends Base {
```

Comment:
Nice! This seems like a flexible pattern. Just curious, is there a case you have in mind when you think we'll want to make `ProfileHelpers` extend from something other than `Object`? This is a workaround to enable a sort of multiple inheritance right?

Reply:
This is a basic pattern for creating mixins in JS, the idea being that you can add any number of these into other classes. This implementation is based on the one in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Mix-ins (which I think copped it from https://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/). For our specific case right here, it didn't _need_ to be a mixin; the MixpanelPeople and MixpanelGroups classes could have just straight inherited from a base class ProfileHelpers, since that's the only shared code. But that's not the right abstraction; we want this set of methods available to stick onto any class (even one that potentially is inheriting from a different class). (In that case it would look like `class Foo extends ProfileHelpers(SomeOtherBaseClass)`)